File: BMICalculator/app/src/main/java/com/example/bmicalculator/MainActivity.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LinearLayout llmain;
        Button btnCalculate;
        EditText edtWeight, edtHeightFt, edtHeightIn;
        TextView txtResult;

        llmain = findViewById(R.id.llmain);
        edtWeight = findViewById(R.id.edtWeight);
        edtHeightFt=findViewById(R.id.edtHeightFt);
        edtHeightIn=findViewById(R.id.edtHeightIn);
        btnCalculate=findViewById(R.id.btnCalculate);
        txtResult=findViewById(R.id.txtResult);


        btnCalculate.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
            int wt=  Integer.parseInt(edtWeight.getText().toString());
            int ft=  Integer.parseInt(edtHeightFt.getText().toString());
            int in=  Integer.parseInt(edtHeightIn.getText().toString());

            int totalIn = ft*12 +in;

            double totalCm = totalIn*2.53;

            double totalM = totalCm/100;

            double bmi= wt/(totalM*totalM);

            if (bmi>25){
                txtResult.setText("You're Overweight");
                llmain.setBackgroundColor(getResources().getColor(R.color.colorOver));
            }else if (bmi<18){
                txtResult.setText("You're Underweight");
                llmain.setBackgroundColor(getResources().getColor(R.color.colorUnder));

            }else{
                txtResult.setText("You're Healthy");
                llmain.setBackgroundColor(getResources().getColor(R.color.healthy));


            }


            }
        });
    }
}

File: BMICalculatorkt/app/src/main/java/com/example/bmicalculatorkt/MainActivity.kt
package com.example.bmicalculatorkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnCalculator.setOnClickListener {
            when {
                edtWeight.text.toString() != "" && edtHeightFt.text.toString() != "" && edtHeightIn.text.toString() != "" -> {

                    val wt = (edtWeight.text.toString()).toDouble()
                    val htFeet = (edtHeightFt.text.toString()).toDouble()
                    val htInch = (edtHeightIn.text.toString()).toDouble()

                    val totalInch = (htFeet*12) + htInch
                    val totalCM= totalInch*2.54
                    val totalM = totalCM/100

                    val bmi = wt/(totalM*totalM)
                    txtResult.text="${bmi.toInt()}"
                    txtTitle.text="Your BMI"
                    txtTitle.setTextColor(resources.getColor(R.color.white))
                    txtMsg.setTextColor(resources.getColor(R.color.white))

                    when {
                        bmi>25 -> {
                            Toast.makeText(this@MainActivity, R.string.strOverWt, Toast.LENGTH_LONG).show()
                            txtMsg.text = resources.getString(R.string.strOverWt)
                            llMain.setBackgroundColor(resources.getColor(R.color.cRed))
                        }
                        bmi<18 -> {
                            Toast.makeText(this@MainActivity, R.string.strUnderWt, Toast.LENGTH_LONG).show()
                            txtMsg.text = resources.getString(R.string.strUnderWt)
                            llMain.setBackgroundColor(resources.getColor(R.color.cYellow))
                        }
                        else -> {
                            Toast.makeText(this@MainActivity, R.string.strHealthy, Toast.LENGTH_LONG).show()
                            txtMsg.text = resources.getString(R.string.strHealthy)
                            llMain.setBackgroundColor(resources.getColor(R.color.cGreen))
                        }
                    }


                }
                else -> {
                    Toast.makeText(this@MainActivity, "Please fill all the entities", Toast.LENGTH_LONG).show()
                }
            }
        }

    }
}